src/lsst/rsp/startup/services/labrunner/labrunner.py

```diff
@@ -534,6 +536,151 @@ def _merge_pgpass(self) -> None:
+                    self._logger.info(f"Replaced link in {fl!s}: {old}->{new}")
+                    fl.write_text(yaml.dump(flensed, default_flow_style=False))
+                    retval = True
+            except Exception:
+                if fl:
+                    self._logger.exception(f"Failed to read/write {fl!s}")
+                else:
+                    self._logger.exception(f"Failed to load yaml from {files}")
+        return retval
+
+    def _inject_new_proxy(self, tgt: Path) -> None:
+        # Conventional for RSP.
+        parent = tgt.parent
+        try:
+            parent.mkdir(parents=True, exist_ok=True)
+        except FileExistsError:
+            self._logger.exception(
+                f"{tgt!s} exists and is not a directory; aborting"
+            )
+            return
+        newlink = "{JUPYTERHUB_PUBLIC_URL}proxy/{port}/status"
+        goodlink = {"distributed": {"dashboard": {"link": newlink}}}
+        if tgt.exists():
+            try:
+                obj = yaml.safe_load(tgt.read_text())
+                newobj = self._replace_proxy_link(obj, goodlink)
+                if newobj:
+                    tgt.write_text(yaml.dump(newobj, default_flow_style=False))
+                else:
+                    self._logger.warning(f"Removing {tgt!s} because empty")
+                    tgt.unlink()
+            except Exception:
+                self._logger.exception(f"Failed to load {tgt!s}")
+        try:
+            tgt.write_text(yaml.dump(goodlink, default_flow_style=False))
+        except Exception:
+            self._logger.exception(f"Failed to write {tgt!s}")
+
+    def _replace_proxy_link(
+        self,
+        obj: dict[str, Any] | None,
+        goodlink: dict[str, dict[str, dict[str, str]]],
+    ) -> dict[str, Any]:
+        flensed = self._flense_dict(obj)
+        if flensed is None:
+            flensed = goodlink
+        newlink = goodlink["distributed"]["dashboard"]["link"]
+        try:
+            current = flensed["distributed"]["dashboard"]["link"]
+            if current == newlink:
+                self._logger.debug("dask dashboard link is already correct")
+                return flensed  # Nothing to do.
+            self._logger.warning(
+                f"Current link is {current}; replacing with {newlink}"
+            )
+        except KeyError:
+            pass  # Setting doesn't currently exist.
+        except Exception:
+            self._logger.exception(f"Failed to parse {flensed}")
+        flensed.update(goodlink)
+        return flensed
+
+    def _flense_dict(
+        self, obj: dict[str, Any] | None
+    ) -> dict[str, Any] | None:
+        """Recursively walk a dict; any place a null value is found, it
+        and its corresponding key are removed.
+        """
+        if not obj:
+            return None
+        retval: dict[str, Any] = {}
+        keylist = list(obj.keys())
+        for keyent in keylist:
```

Comment:
Very minor detail but perhaps could save a couple lines with:

`for key, val in obj.items()`